homeassistant/components/mqtt/select.py

```diff
@@ -148,27 +160,27 @@ def _setup_from_config(self, config):
             ).async_render_with_possible_json_value,
         }
 
-    def _prepare_subscribe_topics(self):
+    def _prepare_subscribe_topics(self) -> None:
         """(Re)Subscribe to topics."""
 
         @callback
         @log_messages(self.hass, self.entity_id)
-        def message_received(msg):
+        def message_received(msg: ReceiveMessage) -> None:
             """Handle new MQTT messages."""
-            payload = self._templates[CONF_VALUE_TEMPLATE](msg.payload)
-
-            if payload.lower() == "none":
+            payload: ReceivePayloadType | None = self._templates[CONF_VALUE_TEMPLATE](
+                msg.payload
+            )
```

Comment:
This would be prettier on two lines I think
```suggestion
            payload: ReceivePayloadType | None
            payload = self._templates[CONF_VALUE_TEMPLATE](msg.payload)
```